src/compartment.c

```diff
@@ -231,6 +227,7 @@ comp_from_elf(char* filename)
             for (size_t j = 0; j < new_comp->relas_cnt; ++j)
             {
                 new_comp->relas[j] = comp_relas[j].r_offset;
+                // TODO double check
```

Comment:
Double check what?

Reply:
The commented code below, which is to do with relocations in non-static binaries. If we will want to handle dynamic executables, I'm unsure if the uncommented code or the commented code is correct right now (and it would take time to check).

Reply:
Presumably we don't currently execute this code path? Assuming so, I'd prefer an `err` or similar to make sure that we bork as soon as we try executing this probably-incorrect path.

Reply:
Yes, you are correct. Added in https://github.com/capablevms/CHERI-ELF-comp/pull/3/commits/ea243ccdd7344becdf1ae9ce5158bf1f9c929b71.